lnwallet/channel.go

```diff
@@ -8689,18 +8689,23 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64) chainfee.SatPerKWe
 	// exactly why it was introduced to react for sharp fee changes.
 	availableBalance, weight := lc.availableBalance(AdditionalHtlc)
 
-	oldFee := lc.localCommitChain.tip().feePerKw.FeeForWeight(weight)
-
+	currentFee := lc.localCommitChain.tip().feePerKw.FeeForWeight(weight)
 	// baseBalance is the maximum amount available for us to spend on fees.
-	baseBalance := availableBalance.ToSatoshis() + oldFee
-
+	baseBalance := availableBalance.ToSatoshis() + currentFee
+
+	// In case our local channel balance is drained, we make sure we do not
+	// decrease the fee rate below the current fee rate. This could lead to
+	// a scenario where we lower the commitment fee rate as low as the fee
+	// floor although current fee rates are way higher. The maximum fee
+	// we allow should not be smaller then the current fee. The decrease
+	// in fee rate should happen when the mempool reports lower fee levels
+	// rather than us decreasing in local balance.
 	maxFee := float64(baseBalance) * maxAllocation
+	maxFee = math.Max(maxFee, float64(currentFee))
```

Comment:
can we avoid mutation here either by inlining the original maxFee definition or renaming the original maxFee to something like maxFeeProposal?